src/plugins/intel_cpu/src/nodes/rnn.cpp

```diff
@@ -1087,8 +1087,16 @@ void RNN::prepareParams() {
             DnnlExtensionUtils::makeDescriptor(query_weights_md(1)),
             DnnlExtensionUtils::makeDescriptor(query_weights_md(2))
         };
-        prepareMemory(intDescs);
-        wasMemoryPrepared = true;
+
+        if (intDescs[1]->getFormatKind() == dnnl_format_kind_rnn_packed) {
```

Comment:
Did not get it actually. Why do we need to always update weights memory if intDescs[1] is happened to be rnn_packed?

Reply:
The logic is, if we get into prepareParams, that means there is shape change. If a shape changes, that means a new primitive is created or get from cache. A new primitive change will create a new memory for weights.
That's why I need it always there.

Reply:
So, even considering the fact, that weights do not depend on dynamic dimensions, such as batch (N) or sequence_length (T), oneDNN actually somehow uses those dimensions while packing into rnn_packed format? And we have to "repack" weights every time those dimentions are changed?

Reply:
Yes, it is.